dask/array/core.py

```diff
@@ -2433,7 +2432,7 @@ def dot(a, b):
 
 def insert_to_ooc(out, arr, lock=True, region=None):
     if lock is True:
-        lock = Lock()
+        lock = SerializableLock()
```

Comment:
You're probably right, but could you explain why? What will happen if the multiprocessing scheduler will be used with this function?